planning/behavior_path_planner/src/scene_module/lane_change/normal.cpp

```diff
@@ -876,182 +907,189 @@ bool NormalLaneChange::getLaneChangePaths(
-
-    // get lateral acceleration range
-    const auto [min_lateral_acc, max_lateral_acc] =
-      common_parameters.lane_change_lat_acc_map.find(initial_lane_changing_velocity);
-    const auto lateral_acc_resolution =
-      std::abs(max_lateral_acc - min_lateral_acc) / lateral_acc_sampling_num;
-    constexpr double lateral_acc_epsilon = 0.01;
-
-    for (double lateral_acc = min_lateral_acc; lateral_acc < max_lateral_acc + lateral_acc_epsilon;
-         lateral_acc += lateral_acc_resolution) {
-      const auto lane_changing_time = PathShifter::calcShiftTimeFromJerk(
-        shift_length, common_parameters.lane_changing_lateral_jerk, lateral_acc);
-      const double longitudinal_acc_on_lane_changing =
-        utils::lane_change::calcLaneChangingAcceleration(
-          initial_lane_changing_velocity, max_path_velocity, lane_changing_time,
-          sampled_longitudinal_acc);
-      const auto lane_changing_length =
-        initial_lane_changing_velocity * lane_changing_time +
-        0.5 * longitudinal_acc_on_lane_changing * lane_changing_time * lane_changing_time;
-      const auto terminal_lane_changing_velocity =
-        initial_lane_changing_velocity + longitudinal_acc_on_lane_changing * lane_changing_time;
-      utils::lane_change::setPrepareVelocity(
-        prepare_segment, current_velocity, terminal_lane_changing_velocity);
-
-      if (lane_changing_length + prepare_length > dist_to_end_of_current_lanes) {
-        RCLCPP_DEBUG(logger_, "length of lane changing path is longer than length to goal!!");
-        continue;
+      // Check if the lane changing start point is not on the lanes next to target lanes,
+      if (target_length_from_lane_change_start_pose > 0.0) {
+        RCLCPP_DEBUG(
+          logger_, "[only new arch] lane change start getEgoPose() is behind target lanelet!!");
+        break;
       }
 
-      if (is_goal_in_route) {
-        const double s_start =
-          lanelet::utils::getArcCoordinates(target_lanes, lane_changing_start_pose).length;
-        const double s_goal =
-          lanelet::utils::getArcCoordinates(target_lanes, route_handler.getGoalPose()).length;
-        const auto num =
-          std::abs(route_handler.getNumLaneToPreferredLane(target_lanes.back(), direction));
-        const double backward_buffer =
-          num == 0 ? 0.0 : common_parameters.backward_length_buffer_for_end_of_lane;
-        const double finish_judge_buffer = common_parameters.lane_change_finish_judge_buffer;
-        if (
-          s_start + lane_changing_length + finish_judge_buffer + backward_buffer +
-            next_lane_change_buffer >
-          s_goal) {
+      const auto shift_length =
+        lanelet::utils::getLateralDistanceToClosestLanelet(target_lanes, lane_changing_start_pose);
+
+      const auto initial_lane_changing_velocity = prepare_velocity;
+      const auto & max_path_velocity =
+        prepare_segment.points.back().point.longitudinal_velocity_mps;
```

Comment:
```suggestion
      const auto max_path_velocity =
        prepare_segment.points.back().point.longitudinal_velocity_mps;
```

Note: just for standardizing this whenever we got chance.
Since this trivial type, i suggesting to remove the `&`